source/common/upstream/upstream_impl.cc

```diff
@@ -574,7 +574,8 @@ ClusterInfoImpl::ClusterInfoImpl(const envoy::api::v2::Cluster& config,
       cluster_socket_options_(parseClusterSocketOptions(config, bind_config)),
       drain_connections_on_host_removal_(config.drain_connections_on_host_removal()),
       warm_hosts_(!config.health_checks().empty() &&
-                  common_lb_config_.ignore_new_hosts_until_first_hc()) {
+                  common_lb_config_.ignore_new_hosts_until_first_hc()),
+      cluster_type_(config.cluster_type()) {
```

Comment:
This optional is always going to have a value. Is that what you want? I don't think so?